lib/associations/belongs-to-many.js

```diff
@@ -113,7 +113,7 @@ var BelongsToMany = function(source, target, options) {
     this.foreignKeyAttribute = {};
     this.foreignKey = this.options.foreignKey || Utils.camelizeIf(
       [
-        Utils.underscoredIf(this.source.options.name.singular, this.source.options.underscored),
+        Utils.underscoredIf(this.options.as || this.source.options.name.singular, this.source.options.underscored),
```

Comment:
Yes it would be for both ?:M.
I'm not certain it's a good idea, but it basically forces users to define foreignKey every time they define as, which is probably not a bad idea but.